batch/batch/batch.py

```diff
@@ -259,7 +259,32 @@ class Job:
         assert self._state == 'Cancelled' or self._state == 'Pending'
         return None
 
-    async def _mark_job_task_complete(self, task_name, log, exit_code, new_state):
+    async def _read_pod_statuses(self):
+        async def _read_pod_status(jt):
+            pod_status = await db.jobs.get_pod_status(*self.id, jt.name)
+            return jt.name, pod_status
+
+        future_pod_statuses = asyncio.gather(*[_read_pod_status(jt) for idx, jt in enumerate(self._tasks)
+                                               if idx < self._task_idx])
```

Comment:
We should make one call to the database. If I'm reading this right we make as many as three calls to the database, each one looking up the same record, but asking for a different column. Just grab all three and take the non-null ones